bandit/core/tester.py

```diff
@@ -71,6 +87,25 @@ def run_tests(self, raw_context, checktype):
                     if result.test_id == "":
                         result.test_id = test._test_id
 
+                    # don't skip a the test if there was no nosec comment
+                    if nosec_tests_to_skip is not None:
+                        # if the set is empty or the test id is in the set of
+                        # tests to skip, log and increment the skip by test
+                        # count
+                        if not nosec_tests_to_skip or \
+                                (result.test_id in nosec_tests_to_skip):
+                            LOG.debug("skipped, nosec for test %s"
+                                      % result.test_id)
+                            scores['nosecs_by_tests'] += 1
+                            continue
+                        # otherwise this test was not called out explicitly by
+                        # a nosec BXX type comment and should fail. Log and
+                        # increment the failed test count
+                        else:
+                            LOG.debug("uncaught test %s in nosec comment"
+                                      % result.test_id)
+                            scores['failed_nosecs_by_test'] += 1
```

Comment:
So we're tracking "Hey, you told me to ignore B102 but then B103 failed here"? Why?

Reply:
It's sort of one of the main drivers for this PR. While I'm sure it probably doesn't happen often, it's currently possible that someone runs bandit, wants to ignore something that was found, ignores the line with nosec, and moves on. Later another person could introduce a new security vulnerability on the line that was nosec'd, but no one would notice. 

So the idea is that if you specifically call out which tests you are ignoring, as more plugins are added to bandit and as the users code evolves they will know about that other vulnerability.

Reply:
It was for counting the number of tests that failed outside of what was asked to be ignored. I'll remove it.